File: test/lexjs-test.py
#! /usr/bin/env python
import sys, lexjs;



# Execute
def main():
	f = open("test1.js", "rb");
	s = f.read().decode("utf-8");
	f.close();

	lexer = lexjs.Lexer(s);

	while (True):
		t = lexer.get_token();
		if (t is None): break;
		sys.stdout.write("{0:s}\n".format(str(t)));

	return 0;



# Run
if (__name__ == "__main__"): sys.exit(main());

File: test/lexjs-test.js
// Testing with node
(function () {
	"use strict";

	// Import
	var fs = require("fs"),
		lexjs = require("./lexjs");



	// Main
	var main = function () {
		var s = fs.readFileSync("test1.js", {
			encoding: "utf8",
			flag: "r",
		});

		var lexer = new lexjs.Lexer(s),
			t;

		while (true) {
			t = lexer.get_token();
			if (t === null) break;
			process.stdout.write(t.to_string() + "\n");
		}

		return 0;
	};



	// Execute
	if (require.main === process.mainModule) process.exit(main());

}).call(this);


